File: PosHubApi/Dtos/DriverDto.cs
using System.Text.Json.Serialization;

namespace PosHubApi.Dtos
{
    public class DriverDto
    {
        [JsonPropertyName("firstName")]
        public string FirstName { get; set; }

        [JsonPropertyName("lastName")]
        public string LastName { get; set; }

        [JsonPropertyName("phoneNumber")]
        public string PhoneNumber { get; set; }

        [JsonPropertyName("driverReference")]
        public string DriverReference { get; set; }

        [JsonPropertyName("vehicle")]
        public VehicleDto Vehicle { get; set; }
    }
}

File: poshub-spa/src/app/dtos/driverDto.ts
import { VehicleDto } from "./vehicleDto";

export class DriverDto {
  constructor(
    public firstName: string = '',
    public lastName: string = '',
    public phoneNumber: string = '',
    public driverReference: string = '',
    public vehicle: VehicleDto = new VehicleDto()
  ) { }
}
